pkg/apis/resource/fuzzer/fuzzer.go

```diff
@@ -18,23 +18,29 @@ package fuzzer
 
 import (
 	fuzz "github.com/google/gofuzz"
-
+	"k8s.io/apimachinery/pkg/runtime"
 	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/kubernetes/pkg/apis/resource"
 )
 
 // Funcs contains the fuzzer functions for the resource group.
 var Funcs = func(codecs runtimeserializer.CodecFactory) []interface{} {
 	return []interface{}{
-		func(obj *resource.ResourceClaimSpec, c fuzz.Continue) {
-			c.FuzzNoCustom(obj) // fuzz self without calling this function again
+		func(r *resource.DeviceRequest, c fuzz.Continue) {
+			c.FuzzNoCustom(r) // fuzz self without calling this function again
 
-			// Custom fuzzing for allocation mode: pick one valid mode randomly.
-			modes := []resource.AllocationMode{
-				resource.AllocationModeImmediate,
-				resource.AllocationModeWaitForFirstConsumer,
+			// Match defaulter.
```

Comment:
Relinking to comment https://github.com/kubernetes/kubernetes/pull/125488#discussion_r1674816886

@pohly: Hmm, nothing obviously fails when removing the fuzzer for resource.DeviceRequest. I'll ask on Slack.

@thockin: There are times when fuzzing within valid space is useful and there are times when breaking assumptions is useful. We mostly seem to use fuzzing for round-trip, so if the test passes without special-cases, that seems best, I think.

Reply:
I've update the fuzzer a bit with this goal in mind (from the updated code, to be pushed soon):
> The goal is to produce actual, real values with a non-zero probability.
> This kicks in whenever the fuzzer produces a null value,
> but not when fields already contain random data.

So the goal isn't anymore to "match defaulting" as mentioned in other fuzzers.

I can also remove. No strong opinion. Both seems to pass tests.

Reply:
There are flakes when not emulating the defaulting:
```
    roundtrip.go:360: ResourceClaimTemplate: diff:   &resource.ResourceClaimTemplate{
          	TypeMeta:   {},
          	ObjectMeta: {Name: "髒ɖʬ]^ɋzȼƹŗƓW脿[谁", GenerateName: "zƗ鐫", Namespace: "枢puǢȏbqŹ署Aj", SelfLink: "拇]ɤ#Ťʯğ}ÅB", ...},
          	Spec: resource.ResourceClaimTemplateSpec{
          		ObjectMeta: {Name: "KĹKŀ填 <b琭荩z", GenerateName: "SPāʣkŨ|", Namespace: "V釯<c犞s", SelfLink: "懻g鲴绸šɒŠ}Ūȫ铻ň串{ƅ维T襻", ...},
          		Spec: resource.ResourceClaimSpec{
          			Devices: resource.DeviceClaim{
          				Requests: []resource.DeviceRequest{
          					{
          						Name:            `:峱3\QPģ¶ǣʏv戀\H4贫釀Ƶ`,
          						DeviceClassName: "ĢDŨnÿ",
          						Selectors:       nil,
        - 						AllocationMode:  "",
        + 						AllocationMode:  "ExactCount",
          						Count:           -818298062021306627,
          						AdminAccess:     false,
          					},
          				},
          				Constraints: nil,
          				Config:      nil,
          			},
          			Controller: "叱傸u<Śþŭ賷Ȏ4rǢ&",
          		},
          	},
          }
```

So defaulting really has to be replicated in the fuzzer. Leaving a string as blank that then gets replaced by the default leads to the diff above.

Will fix.